### PR TITLE
Update cuda svd to account for limited cuda support. Issue  #2675

### PR DESCRIPTION
**Pull request by rcorbish**
## What changes were proposed in this pull request?
Issue  #2675
Cuda only supports SVD on matrix where M>=N ( more rows than cols )

By transposing the input and swapping and transposing outputs U & VT
we can overcome this limitation.

## How was this patch tested?
Compared results to matlab
Added extra junit test cases ( multiply SVD factors and compared against input )
testing with simple code in a test harness

### CODE REVIEW

File: nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/test/java/jcuda/jcublas/ops/LapackTest.java [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
  */
-@Ignore
-public class LapackTest {
-/*
+@RunWith(Parameterized.class)
+public class LapackTest extends BaseNd4jTest {
+
     java.util.Random rng = new java.util.Random(1230) ;
 
+    public TestPCA(Nd4jBackend backend) {
```
Comment (AlexDBlack) [inline nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/test/java/jcuda/jcublas/ops/LapackTest.java]: Invalid constructor name

File: nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/test/java/jcuda/jcublas/ops/LapackTest.java [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
 /**
- * @author raver119@gmail.com
- * // why i'm set as the author here? that's not my code
+ * @author rcorbish
  */
-@Ignore
-public class LapackTest {
-/*
+@RunWith(Parameterized.class)
+public class LapackTest extends BaseNd4jTest {
```
Comment (AlexDBlack) [inline nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/test/java/jcuda/jcublas/ops/LapackTest.java]: BaseNd4jTest isn't available in these tests? None of the other tests under jcuda.jcublas.ops (parallel to this one) seems to use it either...

Comment (rcorbish) [reply nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/test/java/jcuda/jcublas/ops/LapackTest.java]: I will find out what happened & correct. I must have  an old something in my build area.

**Comment by rcorbish** [comment]
Thank-you for your help. I need to figure out how to make the tests work locally.

**Comment by AlexDBlack** [comment]
Sure, np. Thanks for the fixes!
As for the tests - I'm just running them manually in intellij... you will need to build libnd4j (native + cuda) + use nd4j change-cuda-versions.sh if required, and then run a mvn clean install. You should then be able to run those tests in intellij.

**Pull request merged**